virtualsmartcard/win32/BixVReader/VpcdReader.cpp

```diff
@@ -106,50 +106,65 @@ void VpcdReader::shutdown() {
 	vicc_exit((struct vicc_ctx *) ctx);
 	state=SCARD_ABSENT;
 	ctx = NULL;
-	if (waitRemoveIpr!=NULL) {
+	if (!waitRemoveIpr.empty()) {
 		SectionLocker lock(device->m_RequestLock);
-		if (waitRemoveIpr->UnmarkCancelable()==S_OK)
-			waitRemoveIpr->Complete(HRESULT_FROM_WIN32(ERROR_CANCELLED));
-		waitRemoveIpr=NULL;
+		while (!waitRemoveIpr.empty()) {
+			CComPtr<IWDFIoRequest> ipr = waitRemoveIpr.back();
+			if (ipr->UnmarkCancelable()==S_OK)
+				ipr->Complete(HRESULT_FROM_WIN32(ERROR_CANCELLED));
+			waitRemoveIpr.pop_back();
+		}
 	}
-	if (waitInsertIpr!=NULL) {
+	while (!waitInsertIpr.empty()) {
 		SectionLocker lock(device->m_RequestLock);
-		if (waitInsertIpr->UnmarkCancelable()==S_OK)
-			waitInsertIpr->Complete(HRESULT_FROM_WIN32(ERROR_CANCELLED));
-		waitInsertIpr=NULL;
+		CComPtr<IWDFIoRequest> ipr = waitRemoveIpr.back();

```

Comment:
Copy paste error. Should be `CComPtr<IWDFIoRequest> ipr = waitInsertIpr.back();`